protocol/messenger_communities.go

```diff
@@ -709,10 +709,16 @@ func (m *Messenger) getAccountsToShare(addressesToReveal []string, communityID t
 
 		revealedAddress := gethcommon.HexToAddress(verifiedAccount.Address.Hex())
 		revealedAddresses = append(revealedAddresses, revealedAddress)
+		address := verifiedAccount.Address.Hex()
+		isAirdropAddress := address == airdropAddress
+		if airdropAddress == "" {
+			isAirdropAddress = i == 0
```

Comment:
@jrainville, if the parameter `airdropAddress` was validated to never be the empty string in `EditSharedAddresses` and on `RequestToJoinCommunity` it was validated to not be empty if there are addresses to reveal, then why do we need to fallback to the first available address? Shouldn't we return an error or simply not test this condition if the validation exists?

Maybe I misinterpreted the request validations.

Reply:
Good eye. The code is a bit confusing. It's because we accept two cases:
1. Manually selecting addresses: then we need to have a manually select airdrop address as well
2. No selected addresses: in that case, the community manager will parse the addresses and select the wallet accounts automatically for the user. Therefore, we need to assign one of those accounts as the airdrop address, so we select the first account (this is the line we are commenting on). 

When we edit the shared addresses, then we are in the first case, because that comes from the user wanting to manually select addresses. So in the case of edit, we always want the airdrop address to be given (that's why we validate differently there).

Hopefully I explained clearly. Do you want me to add comments to the lines that confused you?

Reply:
Makes perfect sense @jrainville, thanks :)

At first I thought selecting the first available address could be a problem or be undesirable, but it's fine really.

> Do you want me to add comments to the lines that confused you?

Up to you, but it's clear to me now.